src/WebJobs.Script/Workers/Rpc/Configuration/IWorkerProfileConditionFactory.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Script.Workers
+{
+    internal interface IWorkerProfileConditionFactory
+    {
+        IWorkerProfileCondition CreateWorkerProfileCondition(string type, string name, string expression);
```

Comment:
`CreateWorkerProfileCondition` is still dependent on name and expression. How to remove this dependency? Do we pass the JToken/JObject around as a context to the factory?